modules/util/color.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+package util
+
+import (
+	"math"
+	"strconv"
+	"strings"
+)
+
+// Check similar implementation in web_src/js/utils/color.js and keep synchronization
+
+// Return R, G, B values defined in reletive luminance
+func getLuminanceRGB(channel float64) float64 {
+	sRGB := channel / 255
+	if sRGB <= 0.03928 {
+		return sRGB / 12.92
+	}
+	return math.Pow((sRGB+0.055)/1.055, 2.4)
+}
+
+// Get color as RGB values in 0..255 range from the hex color string (with or without #)
+// TODO: support return of rgba
+func HexToRBGColor(colorString string) (float64, float64, float64, error) {
+	var color uint64
+	var err error
+	var r, g, b float64
+	hexString := colorString
+	if strings.HasPrefix(colorString, "#") {
+		hexString = hexString[1:]
+	}
+	// only support transfer of rgb and rrggbb
+	// if not in this format, use default values 0, 0, 0
+	if len(hexString) != 3 && len(hexString) != 6 {
+		return 0, 0, 0, nil
+	}
+	if len(hexString) == 3 {
+		hexString = hexString[0:1] + hexString[0:1] + hexString[1:2] + hexString[1:2] + hexString[2:3] + hexString[2:3]
+	}
+	color, err = strconv.ParseUint(hexString, 16, 64)
+	if err != nil {
+		return 0, 0, 0, err
```

Comment:
I guess the "err" doesn't make sense now. 

Above: `return 0, 0, 0, nil`, so I think it's fine to ignore any error, only return a default color.